src/us/kbase/workspace/database/provenance/CreditMetadataEntry.java

```diff
@@ -0,0 +1,153 @@
+package us.kbase.workspace.database.provenance;
+
+import com.github.zafarkhaja.semver.Version;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import us.kbase.workspace.database.Util;
+import us.kbase.workspace.database.WorkspaceUser;
+/*
+ * Container class for {@link CreditMetadata}; includes metadata about credit metadata.
+ */
+public class CreditMetadataEntry {
+	private final WorkspaceUser user;
+	private final String creditMetadataSchemaVersion;
+	private final Instant timestamp;
+	private final CreditMetadata creditMetadata;
+
+	private CreditMetadataEntry(
+			final WorkspaceUser user,
+			final String creditMetadataSchemaVersion,
+			final Instant timestamp,
+			final CreditMetadata creditMetadata) {
+		this.user = user;
+		this.creditMetadataSchemaVersion = creditMetadataSchemaVersion;
+		this.timestamp = timestamp;
+		this.creditMetadata = creditMetadata;
+	}
+
+	/**
+	 * Gets the workspace user who submitted this {@link CreditMetadata} record.
+	 *
+	 * @return the user.
+	 */
+	public WorkspaceUser getUser() {
+		return user;
+	}
+
+	/**
+	 * Gets the version of the credit metadata schema used in this record.
+	 *
+	 * The credit metadata schema version is documented in the Credit Engine
+	 * repo at https://github.com/kbase/credit_engine/.
+	 *
+	 * @return the schema version.
+	 */
+	public String getCreditMetadataSchemaVersion() {
+		return creditMetadataSchemaVersion;
+	}
+
+	/**
+	 * Gets the timestamp for the creation of this record.
+	 *
+	 * @return the timestamp.
+	 */
+	public Instant getTimestamp() {
+		return timestamp;
+	}
+
+	/**
+	 * Gets the credit metadata.
+	 *
+	 * @return the metadata.
+	 */
+	public CreditMetadata getCreditMetadata() {
+		return creditMetadata;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(creditMetadataSchemaVersion, creditMetadata, timestamp, user);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		CreditMetadataEntry other = (CreditMetadataEntry) obj;
+		return Objects.equals(creditMetadataSchemaVersion, other.creditMetadataSchemaVersion)
+				&& Objects.equals(creditMetadata, other.creditMetadata)
+				&& Objects.equals(timestamp, other.timestamp)
+				&& Objects.equals(user, other.user);
+	}
+
+	/**
+	 * Builds the {@link CreditMetadataEntry}.
+	 *
+	 * @param user
+	 *                workspace user
+	 * @param creditMetadataSchemaVersion
+	 *                version of the metadata schema being used
+	 *                The credit metadata schema version is documented in the
+	 *                Credit Engine repo at https://github.com/kbase/credit_engine/.
```

Comment:
Wait, I thought we said we were going to check the schema into the workspace repo and that would be the source of truth?

Reply:
Yes -- I will update this comment when it's checked in. Once these objects are finalised I need to check the schema is completely in accordance with the version in the CE repo, and then I'll add a copy of the JSONschema version to this repo and update this comment.

Reply:
personally I'd just update the comment right now now, but up to you